src/python/pants/engine/internals/graph_test.py

```diff
@@ -1100,7 +1099,11 @@ def assert_dependencies_resolved(
         enable_dep_inference: bool = False,
     ) -> None:
         target = self.request_single_product(WrappedTarget, requested_address).target
-        args = ["--dependency-inference"] if enable_dep_inference else []
+        args = (
+            ["--python-infer-imports", "--backend-packages=pants.backend.python"]
+            if enable_dep_inference
+            else []
+        )
```

Comment:
Bump that this is unnecessary. You can delete this line and delete the bool flag `enable_dep_inference`.